server/app/services/mongo_pubsub.rb

```diff
@@ -51,28 +63,59 @@ def send_message(data)
 
   attr_accessor :collection
 
+  finalizer :finalize
+
+  # @return [Array<Subscription>]
+  def self.subscriptions
+    @subscriptions ||= []
```

Comment:
This feels wrong but I get why it's done like this. Storing global state outside of actor is not usually a good idea. Do we have any alternatives?

Reply:
Yeah, it's not entirely in line with the actor philosophy, but I'm not convinced the other approaches would be more viable: 

> EDIT: There are also other alternative approaches to fixing #3206, but I think this is the least disruptive to implement: https://github.com/kontena/kontena/issues/3206#issuecomment-357659898

Reply:
Turning each `MongoPubsub::Subscription` itself into an actor that takes care of re-subscribing might vaugely be more correct, but OTOH I think the actual practical complexity of implementing that would be higher than this class-instance-variable-array + `Celluloid::Future` hack...

Reply:
You could also nominally fix the "global state" by passing in the subscriptions array via the supervisor args, so that the actor gets the same array on each restart, but it would effectively still be the same thing.